plugins/cmdmenu.sma

```diff
@@ -140,26 +146,37 @@ public plugin_precache( )
 			copy( szSound, charsmax(szSound), field2 );
 			copy( sndExt, charsmax(sndExt), ".wav" );
 		}
-		else if ( fieldNums == 3 && field1[0] == 'm' && ( field2[0] == 'p' || field2[0] == 'l' ) )	// .mp3 (mp3 play | mp3 loop)
-		{
-			copy( szSound, charsmax(szSound), field3 );
-			copy( sndExt, charsmax(sndExt), ".mp3" );
+		else 
+		{	
+			if ( fieldNums == 3 && field1[0] == 'm' && ( field2[0] == 'p' || field2[0] == 'l' ) )	// .mp3 (mp3 play | mp3 loop)
+			{
+				copy( szSound, charsmax(szSound), field3 );
+				copy( sndExt, charsmax(sndExt), ".mp3" );
+			}
+			else	
+			{		
+				continue;               // WTH is this sound, drop it.
+			}
 		}
-		else												// WTH is this sound, drop it.
-			continue;
 
 		replace_all( szSound, charsmax(szSound), "\'", "" );								// Strips all ugly (and sometimes useless) \'
 
 		if ( szSound[0] == '/' )
+		{
 				replace( szSound, charsmax(szSound), "/", "" );						// Strip leading slash
+		}
 
-		if ( sndExt[1] == 'm' || ( ! equali( szSound, "vox", 3 ) && ! equali( szSound, "fvox", 4 ) && ! equali( szSound, "barney", 6 ) && ! equali( szSound, "hgrunt", 6 ) ) )
+		if ( sndExt[1] == 'm' || ( ! equali( szSound, "vox", strlen("vox") ) && ! equali( szSound, "fvox", strlen("fvox") ) && ! equali( szSound, "barney", strlen("barney") ) && ! equali( szSound, "hgrunt", strlen("hgrunt") ) ) )
```

Comment:
I don't see much improvement here: one native call + duplicating string. Using directly the length was fine in this context. If it bothers you, then you could simple declare `new const voxIdent[] = "vox";` then using `charsmax(voxIdent)`.